### ISSUE TITLE
Controller/action with and without leading backslash

### ISSUE THREAD

**Issue by aimeos** [orchestral/testbench#90]
There's one problem I haven't figured out how to solve this with testbench: According to your last suggestion, I've removed the leading backslash in the "uses" values of the routes. Then, the tests work fine but the application using the package does not. Without the leading backslash, the controllers are not found:

```exception 'ReflectionException' with message 'Class App\Http\Controllers\Aimeos\Shop\Controller\CatalogController does not exist' in .../aimeos.demo.laravel/vendor/compiled.php```

So this works only in the tests: ```'uses' => 'Aimeos\Shop\Controller\AccountController@indexAction'```
and that only in the application: ```'uses' => '\Aimeos\Shop\Controller\AccountController@indexAction'```

Do you know how I can get application and tests working with the same code?

**Comment by crynobone** [comment]
https://github.com/orchestral/testbench/issues/89#issuecomment-92322741

I mention removing the `\` in the `routes.php`, For the caller you still need to use `\` or it would append to the default app namespace which give you `App\Http\Controllers\Aimeos\Shop\Controller\CatalogController`. 

https://github.com/laravel/framework/blob/5.0/src/Illuminate/Routing/UrlGenerator.php#L547-L561

**Comment by crynobone** [comment]
tl;dr

No `\` in `routes.php`, with `\` when you use `URL::action()`.

**Comment by aimeos** [comment]
OK, but this was already the case until I've changed it back because the package URLs wasn't working any more. The tests worked but not the application:
- https://github.com/aimeos/aimeos-laravel/blob/67b9ade66a7c260cfc482b7ec0121a6e9449e951/src/routes.php
- https://github.com/aimeos/aimeos-laravel/blob/67b9ade66a7c260cfc482b7ec0121a6e9449e951/tests/Controller/CatalogControllerTest.php

The other way round it doesn't work either :-(

**Comment by crynobone** [comment]
```php
<?php

Route::match( array( 'GET', 'POST' ), 'myaccount', array(
	'as' => 'aimeos_shop_account',
	'uses' => 'Aimeos\Shop\Controller\AccountController@indexAction'
));
Route::match( array( 'GET', 'POST' ), 'myaccount/favorite/{fav_action?}/{fav_id?}/{d_prodid?}/{d_name?}/{l_pos?}', array(
	'as' => 'aimeos_shop_account_favorite',
	'uses' => 'Aimeos\Shop\Controller\AccountController@indexAction'
));
Route::match( array( 'GET', 'POST' ), 'myaccount/watch/{wat_action?}/{wat_id?}/{d_prodid?}/{d_name?}/{l_pos?}', array(
	'as' => 'aimeos_shop_account_watch',
	'uses' => 'Aimeos\Shop\Controller\AccountController@indexAction'
));
```

```php
<?php

class AccountControllerTest extends AimeosTestAbstract
{
	public function setUp()
	{
		parent::setUp();
		View::addLocation(dirname(__DIR__).'/fixtures/views');

		Route::group(['prefix' => '{site}'], function() {
			require dirname(dirname(__DIR__)).'/src/routes.php';
			require dirname(dirname(__DIR__)).'/src/routes_account.php';
		});
	}


	public function testActions()
	{
		$response = $this->action('GET', '\Aimeos\Shop\Controller\AccountController@indexAction', ['site' => 'unittest']);

		$this->assertResponseOk();
		$this->assertContains('<section class="aimeos account-history">', $response->getContent());
		$this->assertContains('<section class="aimeos account-favorite">', $response->getContent());
		$this->assertContains('<section class="aimeos account-watch">', $response->getContent());
	}
}
```

```
➜  aimeos-laravel git:master ✗ t tests/Controller/AccountControllerTest.php
PHPUnit 4.6.4 by Sebastian Bergmann and contributors.

Configuration read from /Users/crynobone/aimeos-laravel/phpunit.xml

.

Time: 426 ms, Memory: 22.50Mb

OK (1 test, 4 assertions)
```

**Comment by aimeos** [comment]
Yes, the tests work perfectly in this case but the application won't. If you try to set up an application and add the aimeos-laravel package (the steps are described in https://github.com/aimeos/aimeos-laravel/blob/master/README.md), you will get this error when you enter http://<your public root>/index.php/list in your browser:
```
exception 'ReflectionException' with message 'Class App
\Http\Controllers\Aimeos\Shop\Controller\CatalogController does not exist
```
In the **application** it only works **with leading backslashes**, in the tests only without

**Comment by crynobone** [comment]
> In the application it only works with leading backslashes, in the tests only without

*That is a documentation/implementation bug, not related to Testbench at all.*

Do you know what happen when you include your package routes in `app/Http/routes.php`, you're prepending your package to the application namespace, which is by default `App\Http\Controllers`, no packages should do this, or if you wish you need to handle it on your own.

**Comment by aimeos** [comment]
If this shouldn't be done, how can you overwrite routes otherwise and give them e.g. a prefix?

**Comment by crynobone** [comment]
> how can you overwrite routes otherwise and give them e.g. a prefix?

Learn the in and out of Laravel.

<https://github.com/laravel/laravel/blob/master/app/Providers/RouteServiceProvider.php#L38-L41> work around this code.

**Comment by crynobone** [comment]
Or create a config file, were user can define a prefix. https://github.com/FrozenNode/Laravel-Administrator/blob/master/src/config/administrator.php#L10

**Comment by aimeos** [comment]
I'm still learning but that kind of stuff is documented nowhere.
Your suggestion with the config file was great and I've changed the package to do it this way. Now everything seems to work fine. Thanks again!
